clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -0,0 +1,244 @@
+package io.treeverse.clients
+
+import com.google.protobuf.timestamp.Timestamp
+import io.treeverse.clients.LakeFSContext.{
+  LAKEFS_CONF_API_ACCESS_KEY_KEY,
+  LAKEFS_CONF_API_SECRET_KEY_KEY,
+  LAKEFS_CONF_API_URL_KEY
+}
+import org.apache.hadoop.conf.Configuration
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{SparkSession, _}
+
+import scala.annotation.tailrec
+
+object GarbageCollector {
+
+  case class APIConfigurations(apiURL: String, accessKey: String, secretKey: String)
+
+  def getCommitsDF(runID: String, commitDFLocation: String, spark: SparkSession): Dataset[Row] = {
+    spark.read
+      .option("header", value = true)
+      .option("inferSchema", value = true)
+      .csv(commitDFLocation)
+      .where(col("run_id") === runID)
+  }
+
+  private def getRangeTuples(
+      commitID: String,
+      repo: String,
+      conf: APIConfigurations
+  ): Set[(String, String, String)] = {
+    val location =
+      new ApiClient(conf.apiURL, conf.accessKey, conf.secretKey).getMetaRangeURL(repo, commitID)
+    SSTableReader
+      .forMetaRange(new Configuration(), location)
+      .newIterator()
+      .map(range =>
+        (new String(range.id), range.message.minKey.toStringUtf8, range.message.maxKey.toStringUtf8)
```

Comment:
Thinking about it, this fails for invalid UTF-8 keys.  Per [this Oracle blog](https://blogs.oracle.com/corejavatechtips/overhauling-the-java-utf-8-charset) not all byte sequences are parsed as legal JVM Strings, so this may be an issue.  (It *is* possible to load invalid UTF-8 keys into lakeFS AFAIK, because by contrast the [Go ref](https://golang.org/ref/spec#String_types) defines strings as "_a (possibly empty) sequence of bytes_", without enforcing morality.

Can we keep these as bytes?